### PR TITLE
Implemented check to see if there are a sufficient amount of pokeballs

### PR DESCRIPTION
**Pull request by MichaelGatesDev**
Checks to see if the user has more at least 1 of any pokeball (normal, great, ultra, master) before attempting the encounter. Fires a warning event if the player has no pokeballs and does not continue with the encounter which would result in a flee anyway.

### CODE REVIEW

**Comment by mich3lam** [comment]
it breaks it

**Comment by MichaelGatesDev** [comment]
@mich3lam Thanks for the comment. Can you try again? I've updated the code.

**Comment by mich3lam** [comment]
@MichaelGatesDev still breaks my friend

**Comment by MichaelGatesDev** [comment]
@mich3lam  Damn. I'll start a new account and figure this out. Thanks for the help.

**Comment by MichaelGatesDev** [comment]
@mich3lam Could you try it one more time? I appreciate it.

**Comment by mich3lam** [comment]
Sure, i'll try and get back to u with results @MichaelGatesDev

File: PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs [hunk 1]
```diff
@@ -17,6 +18,18 @@ public static class CatchNearbyPokemonsTask
...
         public static void Execute(Context ctx, StateMachine machine)
         {
+            var invResp = ctx.Client.Inventory.GetInventory();
+            var invRespRes = invResp.Result;
+            if (invRespRes.Success)
+            {
+                var items = invResp.Result.InventoryDelta.InventoryItems;
+                if (items.Select(rawItem => rawItem.InventoryItemData.Item).Any(item => (item.ItemId == ItemId.ItemPokeBall || item.ItemId == ItemId.ItemGreatBall || item.ItemId == ItemId.ItemUltraBall || item.ItemId == ItemId.ItemMasterBall) && item.Count == 0))
+                {
+                    Logger.Write("No pokeballs, ignoring pokemon encounters.", LogLevel.Warning);
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs]: We don't use the Logger from the Logic project anymore, create an event, or fire a WarnEvent

File: PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs [hunk 2]
```diff
@@ -17,6 +18,18 @@ public static class CatchNearbyPokemonsTask
     {
         public static void Execute(Context ctx, StateMachine machine)
         {
+            var invResp = ctx.Client.Inventory.GetInventory();
+            var invRespRes = invResp.Result;
+            if (invRespRes.Success)
+            {
+                var items = invResp.Result.InventoryDelta.InventoryItems;
+                if (items.Select(rawItem => rawItem.InventoryItemData.Item).Any(item => (item.ItemId == ItemId.ItemPokeBall || item.ItemId == ItemId.ItemGreatBall || item.ItemId == ItemId.ItemUltraBall || item.ItemId == ItemId.ItemMasterBall) && item.Count == 0))
```
Comment (yamashi) [inline PoGo.NecroBot.Logic/Tasks/CatchNearbyPokemonsTask.cs]: Doubt this is going to work, pretty much says if any of the pokeball types are 0 then we have no pokeballs

**Pull request closed without merging**